File: Semestru 4/MPP/lab/.net/lab3/lab3/Networking/protocol/Request.cs
using Model.app.domain;

namespace Networking.rpcprotocol
{
	[Serializable]
	public class Request<ID>
	{
		public RequestType? Type { get; set; }
		public Entity<ID>? Entity { get; set; }
		public IEnumerable<Entity<ID>>? Entities { get; set; }
		public string? Message { get; set; }

		public override string ToString() =>
			$"Request{{type={Type}, entity={Entity}, entities={Entities}, errorMessage={Message}}}";
	}
}

File: Semestru 4/MPP/lab/java/lab3/Network/src/main/java/org/example/protocol/Request.java
package org.example.protocol;

import org.example.domain.Entity;
import org.example.services.IObserver;

import java.io.Serializable;
import java.util.List;

public class Request implements Serializable {
    private RequestType Type;
    private Entity Entity;
    private List<Entity> Entities;
    private String Message;

    public RequestType getType() {
        return Type;
    }

    public Request setType(RequestType type) {
        Type = type;
        return this;
    }

    public Entity getEntity() {
        return Entity;
    }

    public Request setEntity(Entity entity) {
        Entity = entity;
        return this;
    }

    public List<Entity> getEntities() {
        return Entities;
    }

    public Request setEntities(List<Entity> entities) {
        Entities = entities;
        return this;
    }

    public String getMessage() {
        return Message;
    }

    public Request setMessage(String message) {
        Message = message;
        return this;
    }

    @Override
    public String toString() {
        return "Request [Type= " + Type + ", Entity= " + Entity + ", Entities= " + Entities + ", Message= " + Message + "]";
    }
}
